### ISSUE TITLE
ExtendsContentIntoTitleBar & SetTitleBar make TitleBar unusable

### ISSUE THREAD

**Issue by IcySnex** [microsoft/microsoft-ui-xaml#6993]
### Describe the bug

I have followed the instructions exactly on https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.window.settitlebar to create a custom TitleBar in a new WinUI3 application.
But the new AppTitleBar does not respond to any user interaction.
Its not movable and the minimize/maximize/close buttons do not work but they will show a hover animation.
This issue only occurs with the Unpackaged type.

Xaml:
```
<Grid>
    <Grid.RowDefinitions>
        <RowDefinition Height="32" />
        <RowDefinition />
    </Grid.RowDefinitions>

    <Grid x:Name="AppTitleBar">
        <Image
            Width="16"
            Height="16"
            Margin="8,0"
            HorizontalAlignment="Left"
            Source="Images/WindowIcon.png" />
        <TextBlock
            x:Name="AppTitleTextBlock"
            Margin="28,0,0,0"
            VerticalAlignment="Center"
            Style="{StaticResource CaptionTextBlockStyle}"
            Text="App title"
            TextWrapping="NoWrap" />
    </Grid>

    <NavigationView Grid.Row="1">
       <!--  Page content  -->
    </NavigationView>
</Grid>
```

C#:
```
public MainWindow()
{
    this.InitializeComponent();
    ExtendsContentIntoTitleBar = true;
    SetTitleBar(AppTitleBar);
}
```

Tested with Microsoft.WindowsAppSDK 1.0.3 and 1.1.0-preview1 on Windows 11 21H2.

### Steps to reproduce the bug

1. Create a new WinUi3 application with visual studio
2. Copy the xaml code from https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.window.settitlebar into the MainWindow.xaml window
3. Set the ExtendsContentIntoTitleBar to true and set the AppTitleBar with SetTitleBar() in the MainWindow constructor
4. Run the application
5. Try to use the TitleBar

### Expected behavior

Expected behavior is that the custom TitleBar is working like a normal TilteBar with all functions like dragging, right clicking, minimizing, maximizing and closing like the documentation says here: https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.window.settitlebar#remarks.

### Screenshots

Video of unpackaged:
https://streamable.com/w1n3c1

Video of packaged:
https://streamable.com/l91et4

### NuGet package version

WinUI 3 - Windows App SDK 1.0

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows 11 (21H2): Build 22000

**Comment by viniciusjarina** [comment]
Hi @IcySnex thank you for reporting this issue.

This is actually a problem with unpackaged application and the way that custom title bar is implemented.

TL;DR DragBarWindow uses a `WS_EX_LAYERED` style as a child Window. On Win32 you can only mix `WS_EX_LAYERED` with `WS_CHILD` after Windows 8.

The solution is simple, [add a compatibility](https://github.com/viniciusjarina/CS_Unpackaged/blob/main/CS_Unpackaged/app.manifest#L7) section to the application manifest, this way the `CreateWindowEx` will not fail, since it know it can combine `WS_EX_LAYERED` with `WS_CHILD`

I've pushed a small application unpackaged + managed with the `app.manifest` here [viniciusjarina/CS_Unpackaged: CS_Unpackaged (github.com)](https://github.com/viniciusjarina/CS_Unpackaged)

**Comment by IcySnex** [comment]
This works fine but this messes with my transparent window.

Since WinUI 3 doesnt actually support transparent windows yet I am using some win32 calls which also uses ```WS_EX_LAYERED```.
These calls include ```GetClientRect (winuser.h)``` and ```FillRect (winuser.h)``` etc.
If you set ```ExtendsContentIntoTitleBar = true``` the ```GetClientRect (winuser.h)``` method is kinda buggy (since it removes the entire TitleBar and creates its own) which breaks the TitleBar in some cases.
If you set ```Microsoft.UI.Windowing.AppWindow.TitleBar.ExtendsContentIntoTitleBar = true``` this doesnt happen (since this just trims everything in the TitleBar except the buttons).

For windows 11 devices I can use ```Microsoft.UI.Windowing.AppWindow.TitleBar.ExtendsContentIntoTitleBar = true``` but sadly this doesnt work on windows 10 (https://docs.microsoft.com/en-us/windows/apps/develop/title-bar?#platform-options).

I guess I cant have both so I will just use the default TitleBar for windows 10 devices as long as WinUI doesnt officially support transparent/blurred windows.

Still thanks for the help!

**Comment by castorix** [comment]
With transparency, the transparent rectangle must be resized (to exclude the caption size)
I tested on Windows 10 (with also a Region to round corners of the "DRAG_BAR_WINDOW_CLASS" window (CreateRoundRectRgn)) : 

![CustomTitleBar_Transparent](https://user-images.githubusercontent.com/22345506/164915161-9ea5c29b-5035-4000-8a82-dea802ec73ff.jpg)

**Comment by riverar** [comment]
@asklar Can you help route this? It's pretty busted.

**Comment by BirukTes** [comment]
> Since WinUI 3 doesnt actually support transparent
@castorix 
Could share the project for that window/app you are showing in the image, how did you get 'see through' transparency on the window?, the colours and border colours on the window handle?

**Comment by knightmeister** [comment]
Just an observation: I've noticed a lot of issues with both packaged and unpackaged apps with a custom titlebar. For example I can't get anything to respond to mouse input when it's in the titlebar (i.e. a button can't be clicked).

**Comment by sigmarsson** [comment]
@viniciusjarina Thanks for the support. Is the following snippet the only tweak to be applied ? For me the window remains unanimated upon any mouse cursor event. even on min/max/close.

```
  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
    <application>
      <!--The ID below indicates app support for Windows 8 -->
      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}"/>
    </application>
  </compatibility>
```

My code-behind : 

```
#if WINDOWS
        private static UIElement AppTitleBar()
        {
            var grid = new Grid();
            var text = new TextBlock();

            try
            {
                text.Text = AppInfo.Current.DisplayInfo.DisplayName;
            }
            catch
            {
                text.Text = $"Unpackaged Dev :: {DateTime.Now.ISODateFormat()}";
            }
            
            text.TextWrapping = TextWrapping.NoWrap;
            text.VerticalAlignment = VerticalAlignment.Center;
            
            grid.Children.Add(text);

            return grid;
        }
#endif
```


```
        protected override void InitializeShell(UIElement shell)
        {
#if WINDOWS
            Window = new Window
            {
                ExtendsContentIntoTitleBar = true
            };

            Window.SetTitleBar(AppTitleBar());
```

**Comment by viniciusjarina** [comment]
@sigmarsson I am not sure what you mean by unanimated. Did you check https://github.com/viniciusjarina/CS_Unpackaged that is also not working for you?

**Comment by sigmarsson** [comment]
@viniciusjarina , srry I phrased in an irregular way. Meant that window neither movable nor min/max/close -able.

I attempted to fetch and run  your "CS_Unpackaged" app but does not start up. WinDBG log tail :


```
ModLoad: 00007ff9`deef0000 00007ff9`df0f1000   C:\Windows\System32\twinapi.appcore.dll
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(544)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FF9A5C25809: (caller: 00007FF9A5C2435F) LogHr(1) tid(750c) 80040010 Object is not in any of the inplace active states
    Msg:[Bootstrap.Intitialize: Scanning packages for Major.Minor=1.1, Tag=preview1, MinVersion=1000.447.2307.0] 
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(673)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FF9A5C26B5C: (caller: 00007FF9A5C2435F) Exception(1) tid(750c) 80070491 There was no match for the specified key in the index.
    Msg:[Enumeration: Major.Minor=1.1, Tag=preview1, MinVersion=1000.447.2307.0] 
(4ec8.750c): C++ EH exception - code e06d7363 (first chance)
(4ec8.750c): C++ EH exception - code e06d7363 (first chance)
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(112)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FF9A5C3323A: (caller: 00007FF8C7732C59) ReturnHr(1) tid(750c) 80070491 There was no match for the specified key in the index.
    Msg:[D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(673)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FF9A5C26B5C: (caller: 00007FF9A5C2435F) Exception(1) tid(750c) 80070491 There was no match for the specified key in the index.
    Msg:[Enumeration: Major.Minor=1.1, Tag=preview1, MinVersion=1000.447.2307.0] 
] 
ModLoad: 00007ff9`b2b60000 00007ff9`b2d8e000   C:\WINDOWS\SYSTEM32\icu.dll
(4ec8.750c): CLR exception - code e0434352 (first chance)
CLR: Managed code called FailFast.
There was no match for the specified key in the index. (0x80070491)
ModLoad: 00007ff9`e7140000 00007ff9`e714a000   C:\WINDOWS\SYSTEM32\VERSION.dll
ntdll!NtWaitForMultipleObjects+0x14:
00007ff9`eae0d8c4 c3              ret

```

SDK: 
```
6.0.202 [C:\Program Files\dotnet\sdk]
7.0.100-preview.3.22179.4 [C:\Program Files\dotnet\sdk]
```

**Comment by sigmarsson** [comment]
@viniciusjarina , effective adding the Windows 8 app support is the only integral part here ? Added to `app.manifest` and referernced like 

`    <ApplicationManifest>app.manifest</ApplicationManifest>`

but the window does not accept mouse inputs.
